### DISCUSSION TITLE
Issues getting Chalo Chalo to work

### DISCUSSION THREAD

**Discussion by TehLinkz** [samuelgr/Xidi#77] [Q&A]
This game will allow xinput to be disabled or enabled and I've tried my testing but can't get any of the configs to work.

DirectInput - tried putting dinput8.dll in game directory and nothing, then swapped it with dinput.dll and nothing.

WinMM - This time all controllers were technically detected but the inputs were shared. (In my example the color are the different players)

1. Controller 1 moves blue and red player
2. Controller 2 moves blue player
3. Controller 3 moves blue player
4. Controller 4 moves blue and yellow player
5. Controller 5 moves blue and purple player
6. Controller 6 moves blue player
7. Controller 7 moves blue player
8. Controller 8 moves blue player

HookModule - This fails to launch with the following error:
`C:\Program Files (x86)\Steam\steamapps\common\Chalo
Chalo\_HookshotLauncher_Chalo_Chalo.exe

Hookshot failed to inject this executable.

Error creating a new 32-bit Hookshot executable process (The
system cannot find the file specified).`

Edit: I tried HookModule again using all the 32 bit files (Hookshot.32.exe, Hookshot.32.dll, HookshotLauncher.32.exe, Xidi.HookModule.32.dll) and this did get past the above error, how ever no matter which other xidi form i used (dinput8.dll, dinput.dll), it acted like my above WinMM without Hookshot.

Any advice would be appreciated, maybe I'm doing something wrong or maybe WinMM can be fixed with a config?

**Comment by samuelgr** [comment]
It appears this game is pretty new - what are you hoping to gain by using Xidi with it? How does the game behave **without** Xidi, and what behavior are you expecting **with** Xidi?

**Reply by TehLinkz** [reply]
I talked with the dev of the game via twitter and he's gonna look into updating the code to add a direct input override, but in case he can't I was looking for a work around. Basically I want to use 8 xbox one controllers to play the game but since they by default are xinput, i can only use 4 controllers max. The dev stated that when they demoed it they only used Xbox 360 controllers which support direct input and xinput by default.

**Reply by samuelgr** [reply]
My guess is the game is binding to each Xbox 360 controller twice - once via XInput and a second time via Xidi virtual controllers. So the game thinks it's two different controllers for two different players, but it's actually one physical device. (Or some permutation of this.)

At the moment, Xidi only supports four physical controller devices too, because it uses XInput to talk to them and hence is subject to the limitation of four controllers via XInput.

For your use case I'm not sure Xidi is the right tool. What you would want to do is *hide* all the XInput controllers so the game can only access them via DirectInput/WinMM.

**Reply by TehLinkz** [reply]
Oh I see, I thought Xidi would force xinput devices to act as directinput devices. Do you know of any projects that might be what I'm looking for? As it stands I guess I either have to wait for dev update of game or buy 8 cheap usb controllers that don't use xinput.

**Reply by samuelgr** [reply]
Off the top of my head I don't know of any projects, but that doesn't mean they don't exist.

At very least you'd want an XInput proxy DLL that hides all XInput devices (and I think some do exist). You might also want something that modifies the device IDs of the Xbox 360 controllers via DirectInput, but this might not be needed because your own testing seems to suggest the game uses WinMM.
